### DISCUSSION TITLE
How can I flash it on rpi pico/convert to uf2?

### DISCUSSION THREAD

**Discussion by nivInsanity** [russhughes/st7789_mpy#92] [Q&A]
Hi,

while trying to flash with esptool error is appearing:
"esptool.py v4.3
Serial port COM4
Connecting......................................

A fatal error occurred: Failed to connect to Espressif device: No serial data received.
For troubleshooting steps visit: https://docs.espressif.com/projects/esptool/en/latest/troubleshooting.html".
No success with ubuntu too, is there any other way to flash it to pico or convert firmware.bin to firmware.uf2?

**Comment by asherevan** [comment]
The Raspberry Pi Pico is not a espressif device. It will not work with esptool.

If you hold down the BOOT button while plugging in the pico, it should show up as a drive on your computer that you can simply drag and drop a .uf2 file to.

Is this with a custom built firmware?

**Comment by asherevan** [comment]
But you SHOULD be able to convert a .bin file to .uf2 with something like this command (replace BINFILE with the name of the file you want to convert, OUTFILE with the output file and CHIP with the chip you are using):

`
esptool.py --chip CHIP merge_bin --format uf2 -o OUTFILE.uf2 0x10000 OUTFILE
`

I learned it from [here](https://docs.espressif.com/projects/esptool/en/latest/esp32/esptool/basic-commands.html#uf2-output-format). (I might be wrong)
